src/graph/planner/ngql/SubgraphPlanner.cpp

```diff
@@ -75,8 +75,11 @@ StatusOr<SubPlan> SubgraphPlanner::nSteps(SubPlan& startVidPlan, const std::stri
   subgraph->setOutputVar(input);
   subgraph->setBiDirectEdgeTypes(subgraphCtx_->biDirectEdgeTypes);
   subgraph->setColNames({nebula::kVid});
-
-  auto* condition = loopCondition(steps.steps() + 1, gn->outputVar());
+  uint32_t maxSteps = steps.steps();
+  if (subgraphCtx_->getEdgeProp || subgraphCtx_->withProp) {
+    ++maxSteps;
+  }
+  auto* condition = loopCondition(maxSteps, gn->outputVar());
```

Comment:
Steps of subgraph executor don't keep same with this?

Reply:
no effect